schrack/integrationpluginschrack.cpp

```diff
@@ -96,6 +96,25 @@ void IntegrationPluginSchrack::setupThing(ThingSetupInfo *info)
 
     CionModbusRtuConnection *cionConnection = new CionModbusRtuConnection(hardwareManager()->modbusRtuResource()->getModbusRtuMaster(uuid), address, this);
 
+    connect(cionConnection, &CionModbusRtuConnection::reachableChanged, thing, [cionConnection, thing](bool reachable){
+        qCDebug(dcSchrack()) << "Reachable state changed" << reachable;
+        if (reachable) {
+            cionConnection->initialize();
+        } else {
+            thing->setStateValue("connected", false);
+        }
+    });
+
+    connect(cionConnection, &CionModbusRtuConnection::initializationFinished, info, [info, cionConnection](bool success){
+        qCDebug(dcSchrack()) << "Initialisation finsihed" << success;
+        if (success) {
+            qCDebug(dcSchrack) << "DIP switche states:" << cionConnection->dipSwitches();
```

Comment:
Can you decide if the setup is possible depending on the DIP switches? We could inform the user to enable some if required here?

Reply:
I'm not entirely sure about that yet. Will be added when clarified with Schrack.